File: java/lsp-01-initial/src/main/java/no/bouvet/solid/lsp/BirdService.java
package no.bouvet.solid.lsp;

import java.util.List;


public class BirdService {

    public void feedThemAll(List<Bird> birds) {
        birds.forEach(bird -> bird.eat());
    }

    public void letThemGo(List<Bird> birds) {
        for (Bird bird : birds) {
            if (bird instanceof Penguin) {
                return;
            }
            bird.fly();
        }
    }
}

File: dotnet/Solid/RefactoredSolid/LSP/Example1/BirdService.cs
using System.Collections.Generic;

namespace RefactoredSolid.LSP.Example1
{
    public class BirdService
    {
        public void FeedThemAll(List<Bird> birds)
        {
            birds.ForEach(bird => bird.Eat());
        }

        public void LetThemGo(List<FlyingBird> flyingBirds)
        {
            flyingBirds.ForEach(bird => bird.Fly());
        }
    }
}
